qiita_db/handlers/analysis.py

```diff
@@ -56,15 +59,34 @@ def get(self, analysis_id):
         -------
         dict
             The contents of the analysis keyed by sample id
+
+        Notes
+        -----
+        This response needed to be broken in chunks because we were hitting
+        the max size of a respose: 2G; based on: https://bit.ly/3CPvyjd
         """
+        chunk_len = 1024 * 1024 * 1  # 1 MiB
+
         with qdb.sql_connection.TRN:
             a = _get_analysis(analysis_id)
             mf_fp = qdb.util.get_filepath_information(
                 a.mapping_file)['fullpath']
-            response = None
             if mf_fp is not None:
                 df = qdb.metadata_template.util.load_template_to_dataframe(
                     mf_fp, index='#SampleID')
-                response = df.to_dict(orient='index')
+                response = dumps(df.to_dict(orient='index'))
 
-        self.write(response)
+                crange = range(chunk_len, len(response)+chunk_len, chunk_len)
+                for i, (win) in enumerate(crange):
+                    chunk = response[i*chunk_len:win]
+                    try:
+                        self.write(chunk)
+                        await self.flush()
+                    except StreamClosedError:
+                        break
+                    finally:
```

Comment:
I _think_ the intention here is for this to be `else`?

Reply:
I literally borrow this code from the link in the notes ...

Reply:
sure but it doesn't seem necessary here as this method essentialy returns right after `StreamClosedError` is caught. i don't really care though, so up to you

Reply:
You are right, I guess the difference is how the original code is creating the chunks vs. how it's done here ... I'll remove.